app/models/manageiq/providers/amazon/agent_coordinator_worker.rb

```diff
@@ -1,8 +1,15 @@
 class ManageIQ::Providers::Amazon::AgentCoordinatorWorker < MiqWorker
   require_nested :Runner
 
+  include PerEmsWorkerMixin
```

Comment:
This is going to do one-per-ems instead of one-per-ems-type, so I think this is the wrong choice.

Reply:
We also have to ensure that there is only 1 *total* in the entire zone, even if smart_proxy is enabled on multiple appliances.  We usually do this via the role by making it zonal, but since you are overloading the smart_proxy role for this and it requires multiple workers for other purposes, we may have to do something clever where the query zone.miq_workers or something.  @agrare Thoughts here on how to implement this?

Reply:
https://github.com/ManageIQ/manageiq/blob/481eec1cbaad1259a080f296b6a46c810f989c60/app/models/miq_vim_broker_worker.rb#L16-L20 is an example where the broker handles it, but we still have 1-per-appliance with the role, so we'll have to be *more* clever.

Reply:
@Fryguy the overwrite method desired_queue_names here always return an array with one element. This will guarantee the worker number returns 1, if there is any amazon cloud managers.

Reply:
Ah..yes, that works too...very clever :+1: